core/src/main/scala/org/apache/spark/scheduler/Pool.scala

```diff
@@ -107,7 +109,7 @@ private[spark] class Pool(
     for (schedulable <- sortedSchedulableQueue) {
       sortedTaskSetQueue ++= schedulable.getSortedTaskSetQueue
     }
-    sortedTaskSetQueue
+    sortedTaskSetQueue.filter(_.isSchedulable)
```

Comment:
minor nit: Move this check to `schedulable.getSortedTaskSetQueue` in the loop ?

Reply:
+1